### DISCUSSION TITLE
AES-256 CBC decrypt is incredibly slow and CPU-intensive with large bytes of data

### DISCUSSION THREAD

**Discussion by loves-bruins** [krzyzanowskim/CryptoSwift#975] [General]
Hello,
We are using CryptoSwift at our company and it has worked well with decrypting (AES-256 CBC) small chunks of data.  However I download a required file for an update.  That file has been encrypted using AES-256 CBC and is about 23 megabytes in size.  CryptoSwift decrypt does work but it pegs CPU and takes about 1 full minute to decrypt when on a high priority thread.  
Any ideas on why it's taking so long?  My guess it should be seconds to decrypt.
/Loren

**Comment by krzyzanowskim** [comment]
if this is a debug build, then it's the nature of the Swift debug builds. Release (optimized) build should be > 1000 times faster. So the first thing is this.

**Comment by loves-bruins** [comment]
It is indeed a debug build!  That is excellent news.  Thank you
